hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCommits.java

```diff
@@ -0,0 +1,437 @@
+import org.apache.iceberg.view.ViewMetadata;
+import org.apache.thrift.TException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+/** Test Hive locks and Hive errors and retry during commits. */
+public class TestHiveViewCommits {
+
+  private static final String VIEW_NAME = "test_iceberg_view";
+  private static final String DB_NAME = "hivedb";
+  private static final Namespace ns = Namespace.of(DB_NAME);
+  private static final Schema SCHEMA =
+      new Schema(
+          5,
+          required(3, "id", Types.IntegerType.get(), "unique ID"),
+          required(4, "data", Types.StringType.get()));
+  private static final TableIdentifier VIEW_IDENTIFIER = TableIdentifier.of(ns, VIEW_NAME);
+
+  @RegisterExtension
+  protected static final HiveMetastoreExtension HIVE_METASTORE_EXTENSION =
+      HiveMetastoreExtension.builder().withDatabase(DB_NAME).build();
+
+  private View view;
+
+  private static HiveCatalog catalog;
+
+  @BeforeAll
+  public static void initCatalog() {
+    catalog =
+        (HiveCatalog)
+            CatalogUtil.loadCatalog(
+                HiveCatalog.class.getName(),
+                CatalogUtil.ICEBERG_CATALOG_TYPE_HIVE,
+                ImmutableMap.of(
+                    CatalogProperties.CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS,
+                    String.valueOf(TimeUnit.SECONDS.toMillis(10))),
+                HIVE_METASTORE_EXTENSION.hiveConf());
+  }
+
+  @BeforeEach
+  public void createTestView() {
+    view =
+        catalog
+            .buildView(VIEW_IDENTIFIER)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(ns)
+            .withQuery("hive", "select * from ns.tbl")
+            .create();
+  }
+
+  @AfterEach
+  public void dropTestView() {
+    catalog.dropView(VIEW_IDENTIFIER);
+  }
+
+  @Test
+  public void testSuppressUnlockExceptions() {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    ViewMetadata metadataV1 = ops.current();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1);
+
+    HiveViewOperations spyOps = spy(ops);
+
+    AtomicReference<HiveLock> lockRef = new AtomicReference<>();
+
+    when(spyOps.lockObject())
+        .thenAnswer(
+            i -> {
+              HiveLock lock = (HiveLock) i.callRealMethod();
+              lockRef.set(lock);
+              return lock;
+            });
+
+    try {
+      spyOps.commit(metadataV2, metadataV1);
+      HiveLock spyLock = spy(lockRef.get());
+      doThrow(new RuntimeException()).when(spyLock).unlock();
+    } finally {
+      lockRef.get().unlock();
+    }
+
+    ops.refresh();
+
+    // the commit must succeed
```

Comment:
shouldn't the assertion then make sure that it has `.containsEntry("k1", "v1")`?

Reply:
`spyOps.commit(metadataV2, metadataV1);` is committing the previous metadata(metadataV1) which doesn't have any properties set.  that's the reason 
```
assertThat(ops.current().properties()).hasSize(0);
```
Ref: https://github.com/apache/iceberg/blob/f460964e78dbbfbe81fd093d436dca80c72e7124/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java#L86